### PR TITLE
[move.iterators] Various (hopefully) editorial fixes

### PR DESCRIPTION
**Pull request by johelegp**
Hopefully the intention is crystal clear and this can be handled editorially.

### CODE REVIEW

**Comment by johelegp** [comment]
Increased the scope of the PR. My first attempt to the fix in the second commit was wrong, so there's some noise.

File: source/iterators.tex [hunk 1]
```diff
@@ -4406,7 +4406,7 @@
 \begin{itemdescr}
 \pnum
 \constraints
-The expression \tcode{x + n} shall be valid and have type \tcode{Iterator}.
+The expression \tcode{x + n} shall be valid.
```
Comment (CaseyCarter) [inline source/iterators.tex]: The intent here is "The expression `\tcode{x.base() + n}` shall be valid and have type `\tcode{Iterator}`." Admittedly it's weird to constrain `n + move_iterator` when `move_iterator + n` isn't constrained; I'll open an LWG issue to constrain the member operators similarly.

Comment (CaseyCarter) [reply source/iterators.tex]: ...and while I'm here, thanks for the incredibly detailed review of and corrections to the Ranges wording.

Comment (johelegp) [reply source/iterators.tex]: > The intent here is "The expression `\tcode{x.base() + n}` shall be valid and have type `\tcode{Iterator}`." Admittedly it's weird to constrain `n + move_iterator` when `move_iterator + n` isn't constrained; I'll open an LWG issue to constrain the member operators similarly.

Right. I hoped that [move.iter.requirements] did that job.

> ...and while I'm here, thanks for the incredibly detailed review of and corrections to the Ranges wording.

Sure :)

File: source/iterators.tex [hunk 2]
```diff
@@ -348,11 +349,32 @@
...
+    constexpr bool operator!=(
+      const move_sentinel<S>& x, const move_iterator<Iterator>& y);
+
+  template<class Iterator, SizedSentinel<Iterator> S>
+    constexpr iter_difference_t<Iterator> operator-(
+      const move_sentinel<S>& x, const move_iterator<Iterator>& y);
+  template<class Iterator, SizedSentinel<Iterator> S>
+    constexpr iter_difference_t<Iterator> operator-(
+      const move_iterator<Iterator>& x, const move_sentinel<S>& y);
+
```
Comment (CaseyCarter) [inline source/iterators.tex]: These operators are very much intentionally hidden friends for the effect on name lookup. This change is not editorial.

Comment (johelegp) [reply source/iterators.tex]: Thank you. I guess that's the case for all the friends added?

Comment (CaseyCarter) [reply source/iterators.tex]: Yes, that is the case.

File: source/iterators.tex [hunk 3]
```diff
@@ -4406,7 +4406,7 @@
 \begin{itemdescr}
 \pnum
 \constraints
-The expression \tcode{x + n} shall be valid and have type \tcode{Iterator}.
+The expression \tcode{x.base() + n} shall be valid and have type \tcode{Iterator}.
 
 \pnum
 \returns \tcode{x + n}.
```
Comment (zygoloid) [inline source/iterators.tex]: This change creates an inconsistency between the constraints and the returned value.

`move_iterator:::operator+(difference_type)` has no constraints, so the expression in question is always valid (even though instantiation of the operator might fail), so presumably we're missing a constraint around [move.iter.nav]p7? Whether or not this expression is SFINAEable is not editorial, and the current specification appears to be normatively wrong, so I think this part needs to go to LWG.

Comment (johelegp) [reply source/iterators.tex]: http://wg21.link/LWG3293 attempts to solve this.

Comment (CaseyCarter) [reply source/iterators.tex]: Thanks for the reminder that I need to fix this issue's P/R.

**Comment by johelegp** [comment]
The former change is handled by https://wg21.link/LWG3293 (P3) and the latter is in the draft.

**Pull request closed without merging**